providers/google.go

```diff
@@ -345,3 +348,23 @@ func (p *GoogleProvider) redeemRefreshToken(ctx context.Context, s *sessions.Ses
 
 	return nil
 }
+
+func (p *GoogleProvider) Authorize(_ context.Context, s *sessions.SessionState) (bool, error) {
+	if len(p.AllowedGroups) == 0 {
+		return true, nil
+	}
+
+	for _, group := range s.Groups {
+		if _, ok := p.AllowedGroups[group]; ok {
+			return true, nil
+		}
+	}
+
+	for _, sa := range p.AuthorizedServiceAccounts {
+		if sa == s.Email {
+			return true, nil
+		}
+	}
```

Comment:
I think the issue is broader than service (M2M) accounts. If could also be passing in an bearer token from a person using Retool for example. What should really happen here is that `Authorize` should hook into the special way that groups are retrieved for Google.